### ISSUE TITLE
Internal annotation on setWeight method

### ISSUE THREAD

**Issue by sirkalmi** [caduandrade/multi_split_view#17]
I use the setWeight method of the MultiSplitViewController, but it has an @internal annotation, what is the reason?

**Comment by caduandrade** [comment]
Hi! 

This method should only be used when dragging the dividers.

As it is used internally and in conjunction with another method, it does not notify the Widget or the Listener (if it has been set). Most importantly, it represents an intermediate state. When changing the weight of a child, it is necessary to decrease or increase the weight of the near child. This method is called twice and at the end, the Widget and Listener are notified.

You can set the weight in the controller construction. Are you needing to change this weight after construction?

Unfortunately, Dart doesn't have something similar with Java's access visibility as a package method (if it exists I don't know). So I used @internal.

**Comment by sirkalmi** [comment]
"You can set the weight in the controller construction. Are you needing to change this weight after construction?"

Yes. I want to change the weights at the touch of a button. I want to hide / show the bottom of a split view.

**Comment by caduandrade** [comment]
To hide, we can create a method to set the weight or content of one of the sides to zero.

What are you thinking for the method that shows again? Are you keeping the oldest 2 weights and setting them again? Are you just setting the weight to .5?

**Comment by sirkalmi** [comment]
In the meantime, I realized there was an easier way to do what I wanted.
If I want to maximize one of the views in the split view, I redraw the screen to make the split view disappear. When I want to see the split view again, I can restore the original view. Nevertheless, it may be useful in other cases if these methods are public.

Somehow this is what I did:
builder: (controller) {
  switch (controller.assetListPositionState) {
    case AssetListPositionState.COLLAPSED:
      return MultiSplitView(
        children: [
          MapView(),
          AssetList(),
        ],
      );
    case AssetListPositionState.FULL:
      return AssetList();
    case AssetListPositionState.HIDDEN:
      return MapView();
  }

Thank you for taking the time to find a solution!
